internal/pkg/service/stream/storage/level/local/diskwriter/network/router/assignment/assign.go

```diff
@@ -0,0 +1,30 @@
+// Package assignment provides proportionally assignment of the opened sink slices to source nodes.
+package assignment
+
+import (
+	"math"
+	"slices"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
+)
+
+func AssignSlices(all []model.SliceKey, nodes []string, nodeID string, minSlicesPerNode int) (assigned []model.SliceKey) {
+	// Calculate the average number of slices per source node of the same type.
+	// The value is rounded up.
+	slicesCount := len(all)
+	avgSlicesPerNode := int(math.Ceil(float64(slicesCount) / float64(len(nodes))))
```

Comment:
I would probablly do the if condition to prevent unnecessary panic.

Reply:
> Is there a possibility for nodes to be empty? It would trigger division by zero.

No, the initialization contains "self discovery" from etcd, if no node is found, it fails.

Reply:
We can rename the "must" method here, send a PR